### ISSUE TITLE
Differential backup

### ISSUE THREAD

**Issue by groupsky** [DevJackSmith/mh-hunt-helper#61]
It would be useful to provide differential backups - i.e. only the changes from previous backup. This is most important for the `hunt*` tables.

May be related to #60.

**Comment by DevJackSmith** [comment]
not sure that there's an easy way to do this. I also need to find another place to store backups, github is not a good place for them.

**Comment by groupsky** [comment]
`mysqldump` has [differential/incremental backups](https://dev.mysql.com/doc/refman/5.7/en/backup-methods.html#idm140430566853216) when binary logs are enabled. I haven't used this feature, so it may have some quirks.

Alternatively a custom export script could be used to select all hunts after the last hunt id from previous backup.

Either way it's a low priority feature so until someone has time and interest to implement it, or a simple solution arise this can be delayed.

**Comment by DevJackSmith** [comment]
@tsitu wants this too.

**Comment by DevJackSmith** [comment]
after talking to Tsitu, we wonder how this should be done. First of all there's storage and performance hit if I turn on binary logging, not sure how much, but that might be a stopper. But lets say it's not noticeable. In addition to what we have now, would we do a monthly full back up with daily incremental ones after that? or how? (By the way I completely erased history on repo, so you might have to reclone the repo, it's tiny now, no more back ups in repo.)

**Comment by DevJackSmith** [comment]
also added backup scripts that i currently have in /DB folder, so you can see how I am doing it. (they run on cron)

**Comment by groupsky** [comment]
I just found that `mysqldump` has a [--where](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html#option_mysqldump_where) option that can filter the rows to dump. 

So something like this should do the trick for incremental backups
```bash
last_hunt_id = `cat last_backup_hunt_id`
max_hunt_id = `mysql -e 'select max(id) from hunts'`
mysqldump --where "id > $last_hunt_id AND id <= $max_hunt_id" hunts
mysqldump --where "hunt_id > $last_hunt_id AND hunt_id <= $max_hunt_id" hunt_details
mysqldump --where "hunt_id > $last_hunt_id AND hunt_id <= $max_hunt_id" hunt_loot
mysqldump --where "hunt_id > $last_hunt_id AND hunt_id <= $max_hunt_id" hunt_stage
echo $last_hunt_id > last_backup_hunt_id
```

On the periodicity I think a monthly full backup is better than weekly as I usually update my db copy once or twice a week, except during events when I update daily.

**Comment by DevJackSmith** [comment]
so what's the goal here? what are we trying to solve exactly?
If it's time, i don't see how downloading and applying 15 updates (on average it will be 15 since there are 30-31 days in a month) will be faster than just downloading and applying 1 big one (download time vs applying time).
If it's bandwidth, there are several other ways we can help with this. So can you please clarify what is the main issue here?

**Comment by DevJackSmith** [comment]
ok sounds good. One thing i noticed, is i do mysqldump by whole database, --where flag is table specific. So that would be table by table?

**Comment by groupsky** [comment]
> ok sounds good. One thing i noticed, is i do mysqldump by whole database, --where flag is table specific. So that would be table by table?

Yes, I think so. Maybe mysqldump supports multiple tables with the same filter, but still will need to split hunt related tables from regular ones. 

I'll try to make a script to restore/update from such a dump to see if there are any other issues. Do you have recommendations as to how to pack the dump - i.e. one zip with multiple sqls or single compressed sql file?
